DOCSP-47554 Update Connect to Databases landing page

jwilson-mdb opened the pull request:
## DESCRIPTION
Adds details re: alternative connection options, connection "use cases", and creating a project with a live database connection.

**Note for Reviewer**
The build log contains many `Page not included in any toctree...` errors
- Unused pages will be removed in [DOCSP-47546](https://jira.mongodb.org/browse/DOCSP-47546)
  - Pages are being left in place for now to facilitate:
    - content reorganization updates
    - creation of redirects before publishing feature branch

## STAGING
https://deploy-preview-210--docs-relational-migrator.netlify.app/database-connections/database-connections/

## JIRA
https://jira.mongodb.org/browse/DOCSP-47554

## BUILD LOG


## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

jeff-allen-mongo commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
-when creating projects and migration jobs. You can use these saved database
-connections for any task that requires connecting to a relational or MongoDB
-database.
+You can save a new database connection:
```
[Suggestion]

Our [style guide](https://www.mongodb.com/docs/meta/style-guide/style/lists/#introductory-text) suggests that we should avoid using a fragment to introduce a list. While what you have here is technically a full sentence, it might be slightly easier to parse if we write this as something like:

```suggestion
To save a new database connection, perform one of these actions:
```

jeff-allen-mongo commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
-database.
+You can save a new database connection:
+
+- When creating a project from a live database connection
+
+- When creating or modifying a data migration job
+
+- From the :guilabel:`Database Connections Bar` in a project
+
+- From the Relational Migrator :guilabel:`Connections` page
```
```suggestion
- Create a project from a live database connection.

- Create or modify a data migration job.

- Navigate to the :guilabel:`Database Connections Bar` in a project.

- Navigate to the Relational Migrator :guilabel:`Connections` page.
```

jeff-allen-mongo commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
+
+- From the Relational Migrator :guilabel:`Connections` page
+
+After saving a database connection, you can use it for any 
```
```suggestion
After you save a database connection, you can use it for any 
```

jeff-allen-mongo commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
+Supported Alternatives
+----------------------
+
+Relational Migrator can use Data Definition Language (DDL) files 
+exported from your relational database to create a project when 
+network connectivity or permissions to your relational database 
+are a consideration.
```
[Suggestion]

Start with the condition, so users can tell whether they should care about this information. As in something like:

"If network connectivity or relational database permissions are a consideration, you can use Data Definition Language (DDL) files exported from your relational database to create a Relational Migrator project."

jeff-allen-mongo commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
+The sample schema allows you to try Relational Migrator even if you aren't
+able to connect to a relational database.
+
+These options allow you to create projects and map schemas without live 
```
```suggestion
These options let you create projects and map schemas without live 
```

jeff-allen-mongo commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
+are a consideration.
+
+Relational Migrator also provides a pre-defined relational database schema.
+The sample schema allows you to try Relational Migrator even if you aren't
+able to connect to a relational database.
```
```suggestion
The sample schema lets you try Relational Migrator even if you can't connect to a relational database.
```

jeff-allen-mongo commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
+The sample schema allows you to try Relational Migrator even if you aren't
+able to connect to a relational database.
+
+These options allow you to create projects and map schemas without live 
```
I feel like this paragraph works best as an introduction to this section. At first I had some trouble understanding why I might use these alternative approaches, and I feel like my questions were cleared up here. I'm kind of imagining something like:

"If you don't have a live connection to your relational database, you can use the following options to create projects and map schemas:"

Then a list of the options. What do you think?

jwilson-mdb replied:
I think making this change may also allow me to split the other methods into bullet points vs the current less parseable paragraphs.

jeff-allen-mongo reviewed (changes requested):
@jwilson-mdb thanks for these updates! Some suggestions and considerations for you in the comments.

jwilson-mdb commented on source/database-connections/database-connections.txt:
```diff
@@ -19,21 +19,42 @@ You can save this information as a database connection.
-when creating projects and migration jobs. You can use these saved database
-connections for any task that requires connecting to a relational or MongoDB
-database.
+To save a new database connection, perform one of these actions:
```
Yeah, I think it read more like "use cases" using the old non-style guide approved language (You can do X while Y, Z, etc).

Now it seems more like "Get Started" to me as well, which more closely matches the style guide concept page prototype. (I think that means the style guide is working haha)

I'll update the section title and remove the existing "Get Started" section which I don't think adds much currently.

davidsoulmdb commented:
Nice, I do have some optional suggestions:
1. I would like the individual Oracle, SQL Server connection guides to be front and center. My goals was that these would be the starting point in the actual doc ie a large in-page bullet list of jumping off points?
2. Supported Alternatives is great can we revise the title to state the use case eg 'Using Relational Migrator Without A Live Database Connection' I would also like it to embed steps to importing a DDL file (these are pretty short) or using a demo. I believe these could be migrated into this page.
3. The information on saved connections useful to have written somewhere but not critical to connecting, can we deprioritise eg move to the bottom of the page? It feels like the connections and projects content could be bundled into the bottom of this page entirely with no subpages? I believe the sample ToC didnt have any so am assuming this will be the end goal. Eg just have two titles Managing Database Connections and Managing Migration Projects that lists these operations as a blurb without further subtitles? Goal is to have them available for reference but after the actual connection steps.
PS I am ignoring all these blank pages until cleanup presumably removes them entirely.

jwilson-mdb commented:
Thanks for the feedback @davidsoulmdb !
> 1. I would like the individual Oracle, SQL Server connection guides to be front and center. My goals was that these would be the starting point in the actual doc ie a large in-page bullet list of jumping off points?

I updated the page to match this request. The `Get Started` section now provides a list of links to source DB connection pages and a link to the MongoDB (destination) connection page.

> 2. Supported Alternatives is great can we revise the title to state the use case eg 'Using Relational Migrator Without A Live Database Connection' I would also like it to embed steps to importing a DDL file (these are pretty short) or using a demo. I believe these could be migrated into this page.

I updated the title to `Use Relational Migrator Without a Live Database Connection`. However, adding the embedded steps here would go against our style guidelines regarding [information types](https://www.mongodb.com/docs/meta/style-guide/information-types/index/), which we should avoid. 

> 3. The information on saved connections useful to have written somewhere but not critical to connecting, can we deprioritise eg move to the bottom of the page? It feels like the connections and projects content could be bundled into the bottom of this page entirely with no subpages? I believe the sample ToC didnt have any so am assuming this will be the end goal. Eg just have two titles Managing Database Connections and Managing Migration Projects that lists these operations as a blurb without further subtitles? Goal is to have them available for reference but after the actual connection steps.

The [draft ToC](https://docs.google.com/spreadsheets/d/1fqFkBsiYmoVnm6UAryhFdgVngjFmLT8iKSkgnSmf9k0/edit?gid=292936768#gid=292936768) had dedicated pages for both managing db connections and migration projects. Similar to `#2`, adding procedures to this page would go against the writing team's style guide and best practices.

If you'd like to simplify the Connections L1, we could create a new `Create a Project` L1 above `Connect to Databases` so the ToC would be
```
> Installation
       ...

> Create a Project
       - Manage Projects
       - Project Settings

> Connect to Databases
       - Connect to MySQL
       - Connect to Oracle
       ...
```
Let me know what you think

jwilson-mdb commented:
LGMT'd by @davidsoulmdb in review session
